services/api/src/resources/deployment/resolvers.js

```diff
@@ -719,7 +724,7 @@ const deployEnvironmentPromote = async (
   const sourceEnvironment = R.prop(0, activeEnvironments);
 
   await hasPermission('environment', 'view', {
-    project: sourceEnvironment.project,
+    project: sourceEnvironment.project.id,
```

Comment:
```suggestion
    project: sourceEnvironment.project
```
a bit too over eager replacing :)

Reply:
D'oh! Thanks!